### PR TITLE
Adds an additional optional property to the Databricks WorkspaceEncryptionParameter object.

### PR DESCRIPTION
**Pull request by ashwnacharya**
This property lets the customer encrypt the databricks workspace notebooks at rest.
The customer can specify the key using which the notebook can be encrypted.
This key is fetched from the keyvault using the workspace's system assigned identity.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet: 01-31-2021
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. 02-15-2021
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json [hunk 1]
```diff
@@ -943,6 +947,51 @@
...
+            "keyVersion": {
+              "type": "string",
+              "description": "The version of KeyVault key.",
+              "x-ms-client-name": "KeyVersion"
+            }
+          },
+          "required": [
+            "keyVaultUri",
+            "keyName"
+          ]
```
Comment (mvvsubbu) [inline specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:988]: Validate adding required properties is correct with Feng.

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json [hunk 2]
```diff
@@ -943,6 +947,51 @@
       },
       "description": "The object that contains details of encryption used on the workspace."
     },
+    "ManagedServicesEncryptionEntities": {
```
Comment (mvvsubbu) [inline specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:950]: All the changes looks good

**Review by mvvsubbu — APPROVED** [review]

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json [hunk 3]
```diff
@@ -904,6 +904,10 @@
           "$ref": "#/definitions/WorkspaceCustomParameterType",
           "description": "The type of variable that this is"
         },
+        "entities": {
+          "$ref": "#/definitions/ManagedServicesEncryptionEntities",
```
Comment (mvvsubbu) [inline specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:908]: We need to add a node under entities for "managedServices"

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json [hunk 4]
```diff
@@ -904,6 +904,10 @@
...
+          "description": "The value which should be used for this field"
```
Comment (mvvsubbu) [inline specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:909]: You need to add identity information as well.

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json (hunk 3 above)
Comment (ashwnacharya) [reply specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:908]: Fixed.

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json (hunk 4 above)
Comment (ashwnacharya) [reply specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:909]: Are you sure? There is no identity field in the Milestone 1 spec in the design document.

Comment (mvvsubbu) [reply specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:909]: Yes get the review approved for M5 as well, we can merge the PR once both M1 and M5 changes are deployed. Having to get approval second time is tough

**Review by mvvsubbu — APPROVED** [review]

**Review by mvvsubbu — CHANGES_REQUESTED** [review]
Add the identity changes as well so that it be approved in single review

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json (hunk 4 above)
Comment (mvvsubbu) [reply specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:909]: Ashwin today we discussed and decided we will try to ship the Public Preview scope related stuff in current version and all the MI related stuff in new API version. As per the discussion, we shouldn't have identity in this version, please remove it.

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json [hunk 5]
```diff
@@ -943,6 +947,83 @@
...
+      "properties": {
+        "managedServices": {
+          "description": "Encryption details for managed services",
+          "type": "object",
+          "properties": {
+            "keySource": {
+              "type": "string",
+              "description": "The encryption keySource (provider). Possible values (case-insensitive):  Default, Microsoft.Keyvault",
+              "enum": [
+                "Default",
```
Comment (mvvsubbu) [inline specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:960]: As per our email thread we shouldn't allow "Default", so remove it.

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json [hunk 6]
```diff
@@ -943,6 +947,83 @@
...
+                "Microsoft.Keyvault"
+              ],
+              "x-ms-enum": {
+                "name": "KeySource",
+                "modelAsString": true
+              },
+              "default": "Default"
```
Comment (mvvsubbu) [inline specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:967]: Is the "default" for enum is must, if not lets not add it.

File: specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json [hunk 7]
```diff
@@ -943,6 +947,83 @@
...
+              "type": "object",
+              "properties": {
+                "type": {
+                  "type": "string",
+                  "description": "The type of identity to be used.",
+                  "enum": [
+                    "None",
+                    "UserAssigned"
+                  ],
+                  "x-ms-enum": {
```
Comment (mvvsubbu) [inline specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:1005]: as per latest discussion remove identity from changes

**Review by mvvsubbu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**